packages/console/src/hooks/use-ui-languages.ts

```diff
@@ -0,0 +1,37 @@
+import { builtInLanguages as builtInUiLanguages } from '@logto/phrases-ui';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+
+import { CustomPhraseResponse } from '@/types/custom-phrase';
+
+import { RequestError } from './use-api';
+
+const useUiLanguages = () => {
+  const {
+    data: customPhraseList,
+    error,
+    mutate,
+  } = useSWR<CustomPhraseResponse[], RequestError>('/api/custom-phrases');
+
+  const languages = useMemo(
+    () =>
+      [
+        ...new Set([
+          ...builtInUiLanguages,
+          ...(customPhraseList?.map(({ languageTag }) => languageTag) ?? []),
+        ]),
+      ]
+        .slice()
+        .sort(),
+    [customPhraseList]
+  );
```

Comment:
nit: Would it be better to return empty array when it is still loading? Otherwise, this hook will return the built-in languages first and then the full result set after receiving the response.

Reply:
When it's loading, the SIE will display a skeleton.
If the user's setting is a custom language, it's meaningless to display the built-in languages.
I think we don't need to take care of this unnormal state.